exchanges/kucoin/kucoin.go

```diff
@@ -0,0 +1,1909 @@
+	if loanID == "" {
+		return errors.New("loanId can not be empty")
+	}
+	params["loanId"] = loanID
+	if size == 0 {
+		return errors.New("size can not be zero")
+	}
+	params["size"] = strconv.FormatInt(size, 10)
+	resp := struct {
+		Error
+	}{}
+	return ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, defaultSpotEPL, http.MethodPost, kucoinInitiateIsolatedMarginSingleRepayment, params, &resp)
+}
+
+// GetCurrentServerTime gets the server time
+func (ku *Kucoin) GetCurrentServerTime(ctx context.Context) (time.Time, error) {
+	resp := struct {
+		Timestamp kucoinTimeMilliSec `json:"data"`
+		Error
+	}{}
+	err := ku.SendHTTPRequest(ctx, exchange.RestSpot, defaultSpotEPL, kucoinGetServerTime, &resp)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return resp.Timestamp.Time(), nil
+}
+
+// GetServiceStatus gets the service status
+func (ku *Kucoin) GetServiceStatus(ctx context.Context) (*ServiceStatus, error) {
+	var resp *ServiceStatus
+	return resp, ku.SendHTTPRequest(ctx, exchange.RestSpot, defaultSpotEPL, kucoinGetServiceStatus, &resp)
+}
+
+// PostOrder used to place two types of orders: limit and market
+// Note: use this only for SPOT trades
+func (ku *Kucoin) PostOrder(ctx context.Context, clientOID, side, symbol, orderType, remark, selfTradePrevention, timeInForce string, size, price, cancelAfter, visibleSize, funds float64, postOnly, hidden, iceberg bool) (string, error) {
+	params := make(map[string]interface{})
+	if clientOID == "" {
+		return "", errors.New("clientOid can not be empty")
+	}
+	params["clientOid"] = clientOID
+	if side == "" {
+		return "", errors.New("side can not be empty")
+	}
+	params["side"] = side
+	if symbol == "" {
+		return "", fmt.Errorf("%w, empty symbol", currency.ErrCurrencyPairEmpty)
+	}
+	params["symbol"] = symbol
+	if remark != "" {
+		params["remark"] = remark
+	}
+	if selfTradePrevention != "" {
+		params["stp"] = selfTradePrevention
+	}
+	orderType = strings.ToLower(orderType)
+	switch orderType {
+	case "limit", "":
+		if price <= 0 {
+			return "", errors.New("price can not be empty")
+		}
+		params["price"] = price
+		if size <= 0 {
+			return "", errors.New("size can not be zero or negative")
+		}
+		params["size"] = strconv.FormatFloat(size, 'f', -1, 64)
+		if timeInForce != "" {
+			params["timeInForce"] = timeInForce
+		}
+		if cancelAfter > 0 && timeInForce == "GTT" {
+			params["cancelAfter"] = strconv.FormatFloat(cancelAfter, 'f', -1, 64)
+		}
+		params["postOnly"] = postOnly
+		params["hidden"] = hidden
+		params["iceberg"] = iceberg
+		if visibleSize > 0 {
+			params["visibleSize"] = strconv.FormatFloat(visibleSize, 'f', -1, 64)
+		}
+	case "market":
+		switch {
+		case size > 0:
+			params["size"] = strconv.FormatFloat(size, 'f', -1, 64)
+		case funds > 0:
+			params["funds"] = strconv.FormatFloat(funds, 'f', -1, 64)
+		default:
+			return "", errSizeOrFundIsRequired
+		}
+	default:
+		return "", fmt.Errorf("%w %s", order.ErrTypeIsInvalid, orderType)
+	}
+	if orderType != "" {
```

Comment:
Given that `default` will return an error, this statement is always true and can be removed